serf/serf_test.go

```diff
@@ -1345,9 +1348,9 @@ func TestSerf_Leave_SnapshotRecovery(t *testing.T) {
 		t.Fatalf("err: %s", err)
 	}
 	time.Sleep(s2Config.MemberlistConfig.ProbeInterval * 5)
-
-	// Verify that s2 is "left"
-	testMember(t, s1.Members(), s2Config.NodeName, StatusLeft)
+	// Verify that s2 is not in the memberlist
+	// The join intent is ignored because the leave sleeps long enough for the leave intent to broadcast
+	testMember(t, s1.Members(), s2Config.NodeName, StatusNone)
```

Comment:
@preetapan This doesn't seem right to me. After S2 initiates a leave, it shouldn't disappear as a member until it is reaped. Seems like we expect it to be Left state?

Reply:
This was addressed by changing the test setup to increase ReapInterval in #511